src/mysky.ts

```diff
@@ -181,6 +185,44 @@ export class MySky {
     clearStoredSeed();
   }
 
+  /**
+   * signMessage will sign the given data using the MySky user's private key,
+   * this method can be used for MySky user verification as the signature may be
+   * verified against the user's public key, which is the MySky user id.
+   *
+   * NOTE: verifyMessageSignature is the counter part of this method, and
+   * verifies an original message against the signature and the user's public
+   * key
+   *
+   * NOTE: this function (internally) adds a salt to the given data array to
+   * ensure there's no potential overlap with anything else, like registry
+   * entries.
+   *
+   * @param message - message to sign
+   * @returns signature
+   */
+  async signMessage(message: Uint8Array): Promise<Uint8Array> {
+    // fetch the user's seed
+    const seed = checkStoredSeed();
+    if (!seed) {
+      throw new Error("User seed not found");
+    }
+
+    // fetch the private key
+    const { privateKey } = genKeyPairFromSeed(seed);
+    const privateKeyBytes = fromHexString(privateKey);
+    if (!privateKeyBytes) {
```

Comment:
Can we do a more robust check here? We can at least check whether the length of the private key matches our expectation of 64 bytes.

Reply:
I re-used the `PRIVATE_KEY_LENGTH` and `PUBLIC_KEY_LENGTH` constants from `skynet-js` and verified the length on the `privateKey` variable. Does that work?

Reply:
That's great.